util/package.go

```diff
@@ -452,3 +473,20 @@ func (p *Package) GetDownloadPath() string {
 func (p *Package) GetUrlPath() string {
 	return path.Join("/package", p.Name, p.Version)
 }
+
+// getDatasetTypes return all the dataset types which exist
+// Currently the data is summarised and only the list of
+// datasets is returned, not the counters
+func (p *Package) getDatasetTypes() []string {
+	// Get unique list of
+	typesCounter := map[string]int{}
+	for _, d := range p.DataSets {
+		typesCounter[d.Type] = typesCounter[d.Type] + 1
```

Comment:
I wonder if it doesn't break on the first execution:
`typesCounter[d.Type] = typesCounter[d.Type] + 1`

`typesCounter` is empty, so `typesCounter[d.Type]` would fail.

Reply:
Glad you stumbled over this. I remember thinking exactly the same but then forgetting up it because it works. Need to check in more details.

Reply:
fixed